### PR TITLE
chore(snc): fix SNC errors stemming from `src/compiler/cache.ts`

### PR DESCRIPTION
**Pull request by alicewriteswrongs**
This just fixes a few SNC errors that stem from code in `src/compiler/cache.ts`.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

I think that if unit tests and CI is all passing then we're probably good here

### CODE REVIEW

File: src/testing/mocks.ts [hunk 1]
```diff
@@ -200,14 +200,12 @@ export function mockBuildCtx(config?: Config, compilerCtx?: CompilerCtx): BuildC
 }
 
 export function mockCache(config?: Config, compilerCtx?: CompilerCtx) {
-  if (!config) {
-    config = mockConfig();
-  }
+  const validatedConfig = mockValidatedConfig((config as Partial<ValidatedConfig>) ?? {});
```
Comment (rwaskiewicz) [inline src/testing/mocks.ts:203]: Is the type assertion here necessary? I would have thought TS woulda been happy about this, but maybe I'm missing something!

**Review by rwaskiewicz — COMMENTED** [review]
Two small questions - I'm not entirely convinced they're valid questions, feel free to push back

File: src/testing/mocks.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (alicewriteswrongs) [reply src/testing/mocks.ts:203]: I'm not terribly worried about making the change right now, as it is passing CI and working and whatnot, but I wouldn't be opposed to making the `Config` passed in to `CompilerCtx` and so on a `ValidatedConfig` which would let us do this (pass down a single object that is shared between the `CompilerCtx` and the `Cache`). I started to do so but backed it out in the interest of keeping this diff a bit smaller, but I'll take a stab at it and see how it ends up. Usually swapping the two out gets us a few more SNC violations dropped anyway!

Comment (alicewriteswrongs) [reply src/testing/mocks.ts:203]: yeah it is unfortunately, I thought that the typing of `config` as `d.Config` would satisfy `Partial<ValidatedConfig>` but it did not :/

I think we now have a few fields where the difference between them isn't just that they're optional in `Config` and required on `ValidatedConfig`

File: src/compiler/cache.ts [hunk 2]
```diff
@@ -158,15 +158,12 @@ export class Cache implements d.Cache {
     }
   }
 
-  private getCacheFilePath(key: string) {
+  private getCacheFilePath(key: string): string {
     return join(this.config.cacheDir, key) + '.log';
   }
 
-  getMemoryStats() {
-    if (this.cacheFs != null) {
```
Comment (sean-perkins) [inline src/compiler/cache.ts:166]: There is a number of checks around `cacheFs` being defined. Is it safe to assume it is always defined (meaning we can simplify the other checks) or is it always guaranteed to be defined in the execution when this method would be called?

Comment (alicewriteswrongs) [reply src/compiler/cache.ts:166]: I believe it is guaranteed to be defined, I can't find anywhere where it's reassigned after the initial assignment in the constructor so I think the checks are unnecessary -- I'm going to try removing them and see if anything breaks!

Comment (alicewriteswrongs) [reply src/compiler/cache.ts:166]: nvm did some testing and we need those checks!

File: src/compiler/build/test/build-stats.spec.ts [hunk 3]
```diff
@@ -1,15 +1,16 @@
 import type * as d from '@stencil/core/declarations';
-import { mockBuildCtx, mockCompilerCtx, mockConfig } from '@stencil/core/testing';
+import { mockBuildCtx, mockCompilerCtx, mockValidatedConfig } from '@stencil/core/testing';
 
 import { generateBuildResults } from '../build-results';
 import { generateBuildStats } from '../build-stats';
 
 describe('generateBuildStats', () => {
-  const config = mockConfig();
+  let config: d.ValidatedConfig;
```
Comment (rwaskiewicz) [inline src/compiler/build/test/build-stats.spec.ts:8]: 💯

File: src/compiler/cache.ts (hunk 2 above)
Comment (rwaskiewicz) [reply src/compiler/cache.ts:169]: Does anything need to get pushed back up to GH/changed here if we need these checks?

Comment (alicewriteswrongs) [reply src/compiler/cache.ts:172]: yup, forgot to push reverting that! should be good now

File: src/testing/mocks.ts [hunk 4]
```diff
@@ -230,12 +217,12 @@ export function mockCompilerSystem(): TestingSystem {
   return createTestingSystem();
 }
 
-export function mockDocument(html: string = null) {
+export function mockDocument(html: string) {
```
Comment (tanner-reits) [inline src/testing/mocks.ts:220]: Was the arg here intended not to be optional anymore? Looks like our internal implementations all provide an arg, but it this is also exported from our `testing` module. Also, `mockWindow` below had the same signature and that arg was kept optional.

**Review by tanner-reits — CHANGES_REQUESTED** [review]
One tiny question

File: src/testing/mocks.ts (hunk 4 above)
Comment (alicewriteswrongs) [reply src/testing/mocks.ts:220]: mmm tbh I don't remember why I did this! I'll change it so the type is `string | null`

**Review by tanner-reits — APPROVED** [review]

**Pull request merged**